src/transformers/cache_utils.py

```diff
@@ -955,17 +960,19 @@ def __init__(
         layers = []
         # If a config is passed, use it to infer the layer types and initialize accordingly
         if config is not None:
-            config = config.get_text_config(decoder=True)
-            sliding_window = getattr(config, "sliding_window", None) or getattr(config, "attention_chunk_size", None)
-            layer_types = getattr(config, "layer_types", None)
+            decoder_config = config.get_text_config(decoder=True)
```

Comment:
(no actual change here, except `config` -> `decoder_config` for more descriptive name and easier debugging)